operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java

```diff
@@ -0,0 +1,169 @@
+        // Arrange
+        k8sclient.load(getClass().getResourceAsStream("/example-postgres.yaml")).inNamespace(namespace).createOrReplace();
+        k8sclient.load(getClass().getResourceAsStream("/example-keycloak.yml")).inNamespace(namespace).createOrReplace();
+
+        k8sclient.services().inNamespace(namespace).create(
+                new ServiceBuilder()
+                        .withNewMetadata()
+                        .withName(KEYCLOAK_SERVICE_NAME)
+                        .withNamespace(namespace)
+                        .endMetadata()
+                        .withNewSpec()
+                        .withSelector(Map.of("app", "keycloak"))
+                        .addNewPort()
+                        .withPort(KEYCLOAK_PORT)
+                        .endPort()
+                        .endSpec()
+                        .build()
+        );
+
+        // Act
+        k8sclient.load(getClass().getResourceAsStream("/example-realm.yaml")).inNamespace(namespace).createOrReplace();
+
+        // Assert
+        var crSelector = k8sclient
+                .resources(KeycloakRealmImport.class)
+                .inNamespace(namespace)
+                .withName("example-count0-kc");
+        Awaitility.await()
+                .atMost(3, MINUTES)
+                .pollDelay(5, SECONDS)
+                .ignoreExceptions()
+                .untilAsserted(() -> {
+                    var conditions = crSelector
+                            .get()
+                            .getStatus()
+                            .getConditions();
+
+                    assertThat(getCondition(conditions, DONE).getStatus()).isFalse();
+                    assertThat(getCondition(conditions, STARTED).getStatus()).isTrue();
+                    assertThat(getCondition(conditions, HAS_ERRORS).getStatus()).isFalse();
+                });
+
+        Awaitility.await()
+                .atMost(3, MINUTES)
+                .pollDelay(5, SECONDS)
+                .ignoreExceptions()
+                .untilAsserted(() -> {
+                    var conditions = crSelector
+                            .get()
+                            .getStatus()
+                            .getConditions();
+
+                    assertThat(getCondition(conditions, DONE).getStatus()).isTrue();
+                    assertThat(getCondition(conditions, STARTED).getStatus()).isFalse();
+                    assertThat(getCondition(conditions, HAS_ERRORS).getStatus()).isFalse();
+                });
+
+        String url =
+                "http://" + KEYCLOAK_SERVICE_NAME + "." + namespace + ":" + KEYCLOAK_PORT + "/realms/count0";
+
+        Awaitility.await().atMost(5, MINUTES).untilAsserted(() -> {
+            try {
+                Log.info("Starting curl Pod to test if the realm is available");
+
+                Pod curlPod = k8sclient.run().inNamespace(namespace)
```

Comment:
Although this is an interesting approach, I think it's an avoidable overengineer.....
If you create the service with NodePort instead of ClusterIP you could have a REST api client call to check that using RestAssure

Reply:
Unfortunately this approach won't work "out-of-the-box" in CI, since `minikube` on the Docker driver requires to open a tunnel (e.g. `minikube tunnel`) to access services

Reply:
Maybe tunnel is needed on macOS only?

Reply:
likely

Reply:
In fact it works....at least locally , with Minikube.
Tunnels are only needed for LoadBalancers :
https://minikube.sigs.k8s.io/docs/handbook/accessing/#using-minikube-tunnel

$ kubectl run --image=vad1mo/hello-world-rest --port=5050 hello2 -n
yournamespace
$ kubectl apply -f myservice.yaml -n yournamespace

myservice.yaml ---------------------------
kind: Service
apiVersion: v1
metadata:
name: my-service
spec:
ports:
- protocol: TCP
port: 5050
targetPort: 5050
nodePort: 30007
selector:
run: hello3
type: NodePort

❯ curl $(minikube ip):30007
/ - Hello World! Host:hello3/172.17.0.7%

❯ minikube profile list
|----------|-----------|---------|--------------|------|---------|---------|-------|
| Profile | VM Driver | Runtime | IP | Port | Version | Status
| Nodes |
|----------|-----------|---------|--------------|------|---------|---------|-------|
| minikube | docker | docker | 192.168.58.2 | 8443 | v1.22.3 | Running
| 1 |
|----------|-----------|---------|--------------|------|---------|---------|-------|